test/fluree/db/peer/query_peer_tests.clj

```diff
@@ -0,0 +1,92 @@
+                                          {:headers {"content-type" "application/json"}
+                                           :body    (json/stringify {:db/id "test/test2"})})
+            ledger-list-resp1 @(http/post (str "http://localhost:" query-port "/fdb/dbs"))
+            ledger-list-resp2 @(http/post (str "http://localhost:" ledger-port "/fdb/dbs"))
+            ;; wait for initialization!
+            _                 (Thread/sleep 2000)
+            ledger1-ready?    @(http/post (str "http://localhost:" query-port "/fdb/test/test1/ledger-stats"))
+            ledger2-ready?    @(http/post (str "http://localhost:" ledger-port "/fdb/test/test2/ledger-stats"))]
+        (is (= 200 (:status new-ledger1)))
+        (is (= 200 (:status new-ledger2)))
+        (is (= [["test" "test1"]
+                ["test" "test2"]]
+               (-> ledger-list-resp1 :body json/parse)
+               (-> ledger-list-resp2 :body json/parse)))
+        (is (= "ready" (some-> ledger1-ready? :body json/parse :data :status)))
+        (is (= "ready" (some-> ledger2-ready? :body json/parse :data :status)))))
+
+    (testing "can transact to a ledger"
+      (let [transact-resp1 @(http/post (str "http://localhost:" query-port "/fdb/test/test1/transact")
+                                       {:headers {"content-type" "application/json"}
+                                        :body    (json/stringify [{:_id "_user" :_user/username "query"}])})
+            transact-resp2 @(http/post (str "http://localhost:" ledger-port "/fdb/test/test1/transact")
+                                       {:headers {"content-type" "application/json"}
+                                        :body    (json/stringify [{:_id "_user" :_user/username "ledger"}])})]
+        (is (= 200 (:status transact-resp1)))
+        (is (= 200 (:status transact-resp2)))))
+
+    (testing "can query a ledger"
+      (let [query-query-resp  @(http/post (str "http://localhost:" query-port "/fdb/test/test1/query")
+                                          {:headers {"content-type" "application/json"}
+                                           :body    (json/stringify {:select ["*"] :from "_user"})})
+            ledger-query-resp @(http/post (str "http://localhost:" ledger-port "/fdb/test/test1/query")
+                                          {:headers {"content-type" "application/json"}
+                                           :body    (json/stringify {:select ["*"] :from "_user"})})]
+        (is (= [{:_user/username "ledger"}
+                {:_user/username "query"}]
+               (->> ledger-query-resp
+                    :body
+                    json/parse
+                    (map #(dissoc % :_id)))
+               (->> query-query-resp
+                    :body
+                    json/parse
+                    (map #(dissoc % :_id)))))))
+
+    (stop query-peer)
+    (stop ledger-peer)))
```

Comment:
Yay tests! 🎉 🌮

Reply:
Can you think of any reason running multiple servers on the same JVM would be a bad idea? It's super convenient here, but I'm not 100% sure that it's okay.

Reply:
Seems OK to me! But yeah, there may be dragons in there...